Specify behavior of default this initializer with field initializers (#5940)
dotnet/csharplang: proposals/csharp-10.0/parameterless-struct-constructors.md

Execution of struct instance field initializers matches execution of class field initializers ([§14.11.3](https://github.com/dotnet/csharpstandard/blob/draft-v6/standard/classes.md#14113-instance-variable-initializers)) with **one qualifier**:
> When an instance constructor has no constructor initializer, **or when the constructor initializer `this()` represents the default parameterless constructor**, ... that constructor implicitly performs the initializations specified by the _variable_initializers_ of the instance fields ... . This corresponds to a sequence of assignments that are executed immediately upon entry to the constructor ... . The variable initializers are executed in the textual order in which they appear in the ... declaration.

We also add the following language to [§15.4.9](https://github.com/dotnet/csharpstandard/blob/draft-v6/standard/structs.md#1549-constructors):

> If the struct includes field initializers, then any constructors within the struct which have a `this()` constructor initializer referring to the default parameterless constructor are treated as though they do not have any constructor initializer.

> This means we get an error in a scenario like the following:

Reviewer: Minor: Perhaps "This means an error is reported for the following:"

Author: ```suggestion
This means an error is reported for the following:
```
